File: worker/src/main/java/ru/variantum/worker/repository/ExportEventRepository.java
package ru.variantum.worker.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.query.Param;
import ru.variantum.worker.domain.ExportEventRecord;

import java.util.List;
import java.util.UUID;

public interface ExportEventRepository extends JpaRepository<ExportEventRecord, UUID> {

    List<ExportEventRecord> findTop20ByOrderByReceivedAtDesc();

    long countByFormat(@Param("format") String format);
}

File: worker/src/main/kotlin/ru/variantum/worker/repository/ExportEventRepository.kt
package ru.variantum.worker.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.Param
import ru.variantum.worker.domain.ExportEventRecord
import java.util.UUID

interface ExportEventRepository : JpaRepository<ExportEventRecord, UUID> {

    fun findTop20ByOrderByReceivedAtDesc(): List<ExportEventRecord>

    fun countByFormat(@Param("format") format: String): Long
}
